src/redux/action-types.js

```diff
@@ -0,0 +1,23 @@
+const asyncActions = [
+  'LOAD_ENTITIES',
+  'LOAD_ENTITY'
+].reduce((acc, curr) => [
+  ... acc,
+  ...[curr, `${curr}_PENDING`, `${curr}_FULFILLED`, `${curr}_REJECTED`]
+], [])
+
+
+export const ACTION_TYPES = [
+    ...asyncActions
```

Comment:
@karelhala the indention of the reduce step here was so that one can add additional strings into the array and have them turned into action types of the same name without having to specify both the name and value (which are same).

i.e. instead of
```js
export const SELECT_ENTITY = 'SELECT_ENTITY';
```

you can do

```js
export const ACTION_TYPES = [
    ...asyncActions,
    'SELECT_ENTITY',
    'CHANGE_SORT',
    ...
```

Reply:
Well with this approach you are kinda loosing some import goodies, it's really great for autogenerated items (async actions), but for static ones, you'd have to import `ACTION_TYPES` and then you'd loose check if `SELECT_ENTITY` is imported or not. If such action is changed or removed you will not be notified on build time, which is one of the main reasons for these constants.

Reply:
@karelhala oh I see. I was not aware of the ES module export limitation